src/login/user-runtime-dir.c

```diff
@@ -262,7 +262,7 @@ static int apply_tmpfs_quota(
                 uint64_t v =
                         (scale == 0) ? 0 :
                         (scale == UINT32_MAX) ? UINT64_MAX :
-                        (uint64_t) ((double) (sfs.f_blocks * sfs.f_frsize) / scale * UINT32_MAX);
+                        (uint64_t) ((double) (sfs.f_blocks * sfs.f_frsize) * scale / UINT32_MAX);
```

Comment:
For safety, maybe better to drop `()` around `sfs.f_blocks` and `sfs.f_frsize`. The product may overflow potentially.
Also, better to divide by `QIF_DWBLKSIZE` here?

Reply:
> For safety, maybe better to drop `()` around `sfs.f_blocks` and `sfs.f_frsize`. The product may overflow potentially.

That should not override. Linux does not support disk sizes above UNT64_MAX and thus the multiplication should remain below that.

But I am not opposed to extra paranoia on that...

Reply:
Yeah, that's a paranoia, but still.